crates/redstone/src/protocol/payload_decoder.rs

```diff
@@ -89,7 +89,7 @@ impl<Env: Environment, C: Crypto> PayloadDecoder<Env, C> {
         count: usize,
         value_size: usize,
     ) -> Result<Vec<DataPoint>, Error> {
-        if count != 1 {
+        if count > DATA_PACKAGES_COUNT_MAX_VALUE || count == 0 {
```

Comment:
final minor: maybe this also could be function: `fn check_data_point_count(count) -> Result<(), Error>`? :D